### PR TITLE
Add -fPIC flag to libyak

### LINKED ISSUE #35: -fPIC compile error
On my system, I ran into the following error when compiling:
```
/usr/bin/ld: libyak.a(device_memory.cpp.o): relocation R_X86_64_PC32 against symbol `_ZN7kfusion4cuda5errorEPKcS2_iS2_' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
```
I'm not sure what caused this, as I've successfully compiled the library in the past. Perhaps a new version of CMake or other libraries involved changed something?

Anyway, going through the generated CMake files I saw that the `-fPIC` flag was only added during the compiling of the `.cu` CUDA files and not for the regular `.cpp` files while creating the `libyak` library which in turn caused the error shown above.

### PR DESCRIPTION
**Pull request by dave992**
Added compile flag for `libyak` to fix #35.

### CODE REVIEW

File: yak/CMakeLists.txt [hunk 1]
```diff
@@ -82,6 +82,7 @@ target_link_libraries(${PROJECT_NAME}
   ${CUDA_CUDA_LIBRARY}
   ${PCL_LIBRARIES}  # BUG: shouldn't need this, but get -fPIC errors if it's not included
   Eigen3::Eigen)
+set_target_properties(${PROJECT_NAME} PROPERTIES COMPILE_FLAGS -fPIC)
```
Comment (schornakj) [inline yak/CMakeLists.txt]: Would it be clearer to write this using a named CMake property vs. the raw flag?
```
 set_target_properties(${PROJECT_NAME} PROPERTIES POSITION_INDEPENDENT_CODE ON) 
```

Comment (dave992) [reply yak/CMakeLists.txt]: Sure, I did realize that was an option as well. I will update the commit.

**Comment by dave992** [comment]
Also removed the now obsolete PCL_LIBRARIES target

**Comment by schornakj** [comment]
Looks good to me!

**Pull request merged**